### PR TITLE
Add support for Bazel boolean flags.

### PR DESCRIPTION
**Pull request by cgrindel**
Closes aspect-build/silo#434.

### CODE REVIEW

File: pkg/aspect/root/flags/noable_bool_test.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	g := NewWithT(t)
+	// From Bazel doc
+	// https://bazel.build/reference/command-line-reference#option-syntax
+	doBoolFlagTest(g, false, true, "--foo")
+	doBoolFlagTest(g, true, false, "--nofoo")
+	doBoolFlagTest(g, false, true, "--foo=yes")
+	doBoolFlagTest(g, true, false, "--foo=no")
+	doBoolFlagTest(g, false, true, "--foo=1")
+	doBoolFlagTest(g, true, false, "--foo=0")
+	doBoolFlagTest(g, false, true, "-f")
```
Comment (f0rmiga) [inline pkg/aspect/root/flags/noable_bool_test.go:49]: Missing failing test for invalid value.

File: pkg/aspect/root/flags/noable_bool.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	flags.AddFlag(noFlag)
+
+	return &result
+}
+
+func boolStr(value bool) string {
+	return fmt.Sprintf("%t", value)
+}
+
+type noableBool struct {
```
[the remark this answers is not in the document]
Comment (cgrindel) [reply pkg/aspect/root/flags/noable_bool.go:79]: The name of the flag is stored in the `pflag.Flag` struct. The `noableBool` is just the `Value` implementation.

File: pkg/aspect/root/flags/noable_bool.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+func (nb *noableBool) Set(value string) error {
+	switch strings.ToLower(value) {
+	case "true", "yes", "1":
+		*nb.value = true
+	case "false", "no", "0":
+		*nb.value = false
+	default:
+		return fmt.Errorf("invalid bool value %s", value)
```
[the remark this answers is not in the document]
Comment (cgrindel) [reply pkg/aspect/root/flags/noable_bool.go:99]: The framework handles it.

File: pkg/aspect/root/flags/noable_bool_test.go (hunk 1 above)
Comment (cgrindel) [reply pkg/aspect/root/flags/noable_bool_test.go:65]: Good catch. I added negative tests. Also, I reworked the logic to ensure that noXXX flags can only accept true.

File: pkg/aspect/root/flags/noable_bool.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	BoolFlagNo    = "no"
+	BoolFlag1     = "1"
+	BoolFlag0     = "0"
+)
+
+func RegisterNoableBool(flags *pflag.FlagSet, name string, value bool, usage string) *bool {
+	return RegisterNoableBoolP(flags, name, "", value, usage)
+}
+
+// Register a boolean flag that supports the Bazel option parsing.
```
Comment (gregmagolan) [inline pkg/aspect/root/flags/noable_bool.go:39]: nit: mention the implementation is case-insensitive so values such as `--foo=YES` and `--foo=True` are valid

File: pkg/aspect/root/flags/noable_bool_test.go (hunk 1 above)
Comment (gregmagolan) [reply pkg/aspect/root/flags/noable_bool_test.go:65]: Some case insensitive tests would be good as well: `--foo=YES` `--foo=True` etc...

Comment (cgrindel) [reply pkg/aspect/root/flags/noable_bool_test.go:67]: Done

File: pkg/aspect/root/flags/noable_bool.go (hunk 4 above)
Comment (cgrindel) [reply pkg/aspect/root/flags/noable_bool.go]: Done

**Comment by cgrindel** [comment]
@gregmagolan @f0rmiga I believe that I have made all of the updates to the PR. Could you take another look?

File: pkg/aspect/root/flags/noable_bool.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	BoolFlagNo    = "no"
+	BoolFlag1     = "1"
+	BoolFlag0     = "0"
+)
+
+func RegisterNoableBool(flags *pflag.FlagSet, name string, value bool, usage string) *bool {
+	return RegisterNoableBoolP(flags, name, "", value, usage)
+}
+
+// Register a boolean flag that supports Bazel option parsing.
```
Comment (f0rmiga) [inline pkg/aspect/root/flags/noable_bool.go:39]: Nit: not a valid godoc comment.

**Review by f0rmiga — APPROVED** [review]

File: pkg/aspect/root/flags/noable_bool.go (hunk 5 above)
Comment (cgrindel) [reply pkg/aspect/root/flags/noable_bool.go:39]: The automerge happened. However, I am willing to fix the issue. I don't see what the problem is with the comment. What would you recommend that I change?

Comment (f0rmiga) [reply pkg/aspect/root/flags/noable_bool.go:39]: ```// RegisterNoableBoolP registers a boolean flag ...```

I wonder why our current linters are not catching those. Maybe I missed some configuration.

Comment (f0rmiga) [reply pkg/aspect/root/flags/noable_bool.go:39]: I also changed the merge rules to require that comments are resolved before auto-merge happens.

**Pull request merged**